apps/C/graphic_screen_test.c

```diff
@@ -0,0 +1,127 @@
+#define WRITE_REG(REG,VAL) (*REG) = VAL
+#define READ_REG(VAL,REG)  VAL    = (*REG)
+#define VGA_PTR(PTR,OFF)   PTR    = (volatile int *) (VGA_MEM_BASE + OFF)
+
+/* Control registers addresses */
+#define CR_MEM_BASE 0x00002000
+#define CR_CURSOR_H (volatile int *) (CR_MEM_BASE + 0x2c)
+#define CR_CURSOR_V (volatile int *) (CR_MEM_BASE + 0x28)
+
+/* VGA defines */
+#define VGA_MEM_BASE       0x00003000
+#define VGA_MEM_SIZE_BYTES 38400
+#define VGA_MEM_SIZE_WORDS 9600
+#define LINE               320
+#define BYTES              4
+#define COLUMN             80 /* COLUMN between 0 - 79 (80x60) */
+#define RAWS               60 /* RAWS between 0 - 59 (80x60) */
+
+/* ASCII table */
+#define H        0x48
+#define E        0x45
+#define L        0x4C
+#define O        0x4F
+/* Letters */
+#define H_TOP    0b00111111001100110011001100000000
+#define H_BOTTOM 0b00000000001100110011001100110011
+
+#define E_TOP    0b00011111000000110011111100000000
+#define E_BOTTOM 0b00000000001111110000001100000011
+
+#define L_TOP    0b00000011000000110000001100000000
+#define L_BOTTOM 0b00000000001111110000001100000011
+
+#define O_TOP    0b00110011001100110001111000000000
+#define O_BOTTOM 0b00000000000111100011001100110011
+
+/* This function print a char note on the screen in (raw,col) position */
+void draw_char(char note, int raw, int col)
+{
+    unsigned int vertical   = raw * 320;
+    unsigned int horizontal = col * 4;
+    volatile int *ptr_top;
+    volatile int *ptr_bottom;
+    VGA_PTR(ptr_top    , horizontal + vertical);
+    VGA_PTR(ptr_bottom , horizontal + vertical + 320);
+
+    switch (note)
```

Comment:
Use an Array in memory and "mux" acording to ascii value

Reply:
Issue opened.